GW2EIEvtcParser/EIData/Statistics/StatisticsHelper.cs

```diff
@@ -1,265 +1,276 @@
+    public IReadOnlyCollection<Buff> GetPresentRemainingBuffsOnPlayer(AbstractSingleActor actor)
+    {
+        if (actor is Player p && _presentRemainingBuffsPerPlayer.TryGetValue(p, out var buffs))
         {
-            if (!(actor is Player p))
-            {
-                return new HashSet<Buff>();
-            }
-            if (_presentRemainingBuffsPerPlayer.TryGetValue(p, out HashSet<Buff> buffs))
-            {
-                return buffs;
-            }
-            return new HashSet<Buff>();
+            return buffs;
         }
+        return [ ];
+    }
+
+    //
+
+    private readonly List<Buff> _presentBoons = new();//Used only for Boon tables
+    private readonly List<Buff> _presentConditions = new();//Used only for Condition tables
+    private readonly List<Buff> _presentOffbuffs = new();//Used only for Off Buff tables
+    private readonly List<Buff> _presentSupbuffs = new();//Used only for Off Buff tables
+    private readonly List<Buff> _presentDefbuffs = new();//Used only for Def Buff tables
+    private readonly List<Buff> _presentDebuffs = new();//Used only for Debuff tables
+    private readonly List<Buff> _presentGearbuffs = new();//Used only for Gear Buff tables
+    private readonly List<Buff> _presentNourishments = new();
+    private readonly List<Buff> _presentEnhancements = new();
+    private readonly List<Buff> _presentOtherConsumables = new();
+    private readonly Dictionary<Player, HashSet<Buff>> _presentRemainingBuffsPerPlayer = new();
 
-        //
 
-        private readonly List<Buff> _presentBoons = new List<Buff>();//Used only for Boon tables
-        private readonly List<Buff> _presentConditions = new List<Buff>();//Used only for Condition tables
-        private readonly List<Buff> _presentOffbuffs = new List<Buff>();//Used only for Off Buff tables
-        private readonly List<Buff> _presentSupbuffs = new List<Buff>();//Used only for Off Buff tables
-        private readonly List<Buff> _presentDefbuffs = new List<Buff>();//Used only for Def Buff tables
-        private readonly List<Buff> _presentDebuffs = new List<Buff>();//Used only for Debuff tables
-        private readonly List<Buff> _presentGearbuffs = new List<Buff>();//Used only for Gear Buff tables
-        private readonly List<Buff> _presentNourishments = new List<Buff>();
-        private readonly List<Buff> _presentEnhancements = new List<Buff>();
-        private readonly List<Buff> _presentOtherConsumables = new List<Buff>();
-        private readonly Dictionary<Player, HashSet<Buff>> _presentRemainingBuffsPerPlayer = new Dictionary<Player, HashSet<Buff>>();
+    //Positions for group
+    private List<ParametricPoint3D?>? _stackCenterPositions = null;
+    private List<ParametricPoint3D?>? _stackCommanderPositions = null;
 
+    //TODO(Rennorb) @cleanup: So ... why are these parametric points, but we still add null values between them?
```

Comment:
Convenience mainly. It guarentees the same delta between elements and easily find start offsets. 
In GetDistanceToTarget, the number of skipped elements on processed actor gives us immediately the start index on the stackCenter/commanderPosition lists.